### PR TITLE
feat: Integrate sengrid

### PR DESCRIPTION
**Pull request by ashikka**

### CODE REVIEW

File: services/email/email.js [hunk 1]
```diff
@@ -1,38 +1,28 @@
...
-        to: ctx.params.to,
-        subject: ctx.params.subject,
-        text: ctx.params.text,
-        html: ctx.params.html
-      }
+
       try {
-        const res = await mailgun.messages().send(data)
-        return (res)
+        if (Math.floor(Math.random() * 2) % 2) {
```
Comment (theProgrammerDavid) [inline services/email/email.js]: Please fix the logic in the `if` statement. Dividing the load between mailgun and sendgrid can be done differently .
You could split the list in half and assign them to Sendgrid and Mailgun

**Review by theProgrammerDavid — COMMENTED** [review]
Alternatively, @thebongy  suggested using `ML` for true load balancing (as stated by @alias-rahil )

`
(Math.floor(Math.random() * 10) >> (Math.floor((Math.random()*2) & Math.floor(Math.random()*2)) | Math.floor(Math.random()) *2) == 1)
`

**Comment by thebongy** [comment]
@theProgrammerDavid 

Alternatively, you could also do

math.random() > 0.5

**Comment by theProgrammerDavid** [comment]
> @theProgrammerDavid
> 
> Alternatively, you could also do
> 
> math.random() > 0.5

But then it wouldn't be `ML`

**Comment by alias-rahil** [comment]
fixed in https://github.com/ashikka/email-service/commit/6f3ec6d649946dfe056e8ba05c95d533615486d2

File: services/email/email.js (hunk 1 above)
Comment (alias-rahil) [reply services/email/email.js]: fixed in https://github.com/ashikka/email-service/commit/6f3ec6d649946dfe056e8ba05c95d533615486d2

**Pull request merged**